### PR TITLE
Issue 1325 change auto asign trnasformers to handle id types

### PR DESCRIPTION
**Pull request by pvk-developer**

### CODE REVIEW

File: sdv/data_processing/data_processor.py [hunk 1]
```diff
@@ -363,42 +364,48 @@ def create_anonymized_transformer(sdtype, column_metadata):
...
-
         else:
-            kwargs = deepcopy(column_metadata)
-            kwargs['enforce_uniqueness'] = True
-            transformer = self.create_anonymized_transformer(sdtype, kwargs)
+            regex_format = column_metadata.get('regex_format', '[0-1a-z]{5}')
+            transformer = rdt.transformers.RegexGenerator(
+                regex_format=regex_format,
+                enforce_uniqueness=(column_name in self._keys)
+            )
```
Comment (frances-h) [inline sdv/data_processing/data_processor.py:408]: nitpick, but could be shortened to:
```python
default_regex_format = r'\d{30} if is_numeric else '[0-1a-z]{5}'
regex_format = column_metadata.get('regex_format', default_regex_format)
transformer = rdt.transformers.RegexGenerator(
    regex_format=regex_format,
    enforce_uniqueness=(column_name in self._keys)
)
```

**Review by frances-h — APPROVED** [review]
Looks good, just one small comment

File: sdv/data_processing/data_processor.py (hunk 1 above)
Comment (pvk-developer) [reply sdv/data_processing/data_processor.py:408]: very pythonic! I like it 👍🏻 💯

File: sdv/datasets/demo.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def _validate_output_folder(output_folder_name):
 
 
 def _download(modality, dataset_name):
-    dataset_url = BUCKET_URL + '/' + modality.upper() + '/' + dataset_name + '.zip'
+    dataset_url = BUCKET_URL + '/' + modality + '/' + dataset_name + '.zip'
```
Comment (amontanez24) [inline sdv/datasets/demo.py:42]: minor: you can do this as f-string

**Review by amontanez24 — APPROVED** [review]
minor comment and question but this looks good!

File: sdv/datasets/demo.py [hunk 3]
```diff
@@ -19,8 +19,8 @@
 from sdv.metadata.single_table import SingleTableMetadata
 
 LOGGER = logging.getLogger(__name__)
-BUCKET = 'sdv-demo-datasets'
-BUCKET_URL = 'https://sdv-demo-datasets.s3.amazonaws.com'
+BUCKET = 'sdv-updated-datasets'
```
[the remark this answers is not in the document]
Comment (pvk-developer) [reply sdv/datasets/demo.py:22]: Yes this has to be 'undone' and probably the `upper()` in the modality.

**Pull request merged**